# Indicator Variable and Interaction Term

Load the dataset named Carseats (in the ISLR package) into R.

 

Background Information

Carseats is a simulated dataset in the ISLR package with sales of child car seats at 400 different stores. The 11 variables are:

str(Carseats)

 

- Sales: Unit sales (in thousands) at each location

- CompPrice: Price charged by competitor at each location

- Income: Community income level (in thousands of dollars)

- Advertising: Local advertising budget for company at each location (in thousands of dollars)

- Population: Population size in region (in thousands)

- Price: Price company charges for car seats at each site

- ShelveLoc: A factor with levels Bad, Good and Medium indicating the quality of the shelving location for the car seats at each site

- Age: Average age of the local population

- Education: Education level at each location

- Urban: A factor with levels No and Yes to indicate whether the store is in an urban or rural location

- US: A factor with levels No and Yes to indicate whether the store is in the US or not

 

In [1]:
library(ggplot2) 
library("tidyverse")
library(Ecdat)
library(ISLR)
str(Carseats)

Warning message:
"package 'ggplot2' was built under R version 3.4.4"Warning message:
"package 'tidyverse' was built under R version 3.4.4"-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
v tibble  1.4.2     v purrr   0.2.5
v tidyr   0.8.1     v dplyr   0.7.6
v readr   1.1.1     v stringr 1.3.1
v tibble  1.4.2     v forcats 0.3.0
Warning message:
"package 'tibble' was built under R version 3.4.4"Warning message:
"package 'tidyr' was built under R version 3.4.4"Warning message:
"package 'readr' was built under R version 3.4.4"Warning message:
"package 'purrr' was built under R version 3.4.4"Warning message:
"package 'dplyr' was built under R version 3.4.4"Warning message:
"package 'stringr' was built under R version 3.4.4"Warning message:
"package 'forcats' was built under R version 3.4.4"-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()
Warning mess

'data.frame':	400 obs. of  11 variables:
 $ Sales      : num  9.5 11.22 10.06 7.4 4.15 ...
 $ CompPrice  : num  138 111 113 117 141 124 115 136 132 132 ...
 $ Income     : num  73 48 35 100 64 113 105 81 110 113 ...
 $ Advertising: num  11 16 10 4 3 13 0 15 0 0 ...
 $ Population : num  276 260 269 466 340 501 45 425 108 131 ...
 $ Price      : num  120 83 80 97 128 72 108 120 124 124 ...
 $ ShelveLoc  : Factor w/ 3 levels "Bad","Good","Medium": 1 2 3 3 1 1 3 2 3 3 ...
 $ Age        : num  42 65 59 55 38 78 71 67 76 76 ...
 $ Education  : num  17 10 12 14 13 16 15 10 10 17 ...
 $ Urban      : Factor w/ 2 levels "No","Yes": 2 2 2 2 2 1 2 2 1 1 ...
 $ US         : Factor w/ 2 levels "No","Yes": 2 2 2 2 1 2 1 2 1 2 ...


Create a new dataframe (call it C1) that is a copy of Carseats. <br>Create two indicator (dummy) variables:

- Bad_Shelf = 1 if ShelveLoc = “Bad”, 0 otherwise
- Good_Shelf = 1 if ShelveLoc = “Good”, 0 otherwise
 


In [2]:
C1<-data.frame(Carseats) %>%
    mutate(Bad_Shelf=ifelse(ShelveLoc=='Bad',1,0)) %>%
    mutate(Good_Shelf=ifelse(ShelveLoc=="Good",1,0))
head(C1,20)


Warning message:
"package 'bindrcpp' was built under R version 3.4.4"

Sales CompPrice Income Advertising Population Price ShelveLoc Age Education
1   9.50 138        73    11          276        120   Bad       42  17       
2  11.22 111        48    16          260         83   Good      65  10       
3  10.06 113        35    10          269         80   Medium    59  12       
4   7.40 117       100     4          466         97   Medium    55  14       
5   4.15 141        64     3          340        128   Bad       38  13       
6  10.81 124       113    13          501         72   Bad       78  16       
7   6.63 115       105     0           45        108   Medium    71  15       
8  11.85 136        81    15          425        120   Good      67  10       
9   6.54 132       110     0          108        124   Medium    76  10       
10  4.69 132       113     0          131        124   Medium    76  17       
11  9.01 121        78     9          150        100   Bad       26  10       
12 11.96 117        94     4          503         94   Good      50  13       
13  3.98 122        35     2          393        136   Medium    62  18       
14 10.96 115        28    11           29         86   Good      53  18       
15 11.17 107       117    11          148        118   Good      52  18       
16  8.71 149        95     5          400        144   Medium    76  18       
17  7.58 118        32     0          284        110   Good      63  13       
18 12.29 147        74    13          251        131   Good      52  10       
19 13.91 110       110     0          408         68   Good      46  17       
20  8.73 129        76    16           58        121   Medium    69  12       
   Urban US  Bad_Shelf Good_Shelf
1  Yes   Yes 1         0         
2  Yes   Yes 0         1         
3  Yes   Yes 0         0         
4  Yes   Yes 0         0         
5  Yes   No  1         0         
6  No    Yes 1         0         
7  Yes   No  0         0         
8  Yes   Yes 0         1         
9  No    No  0         0         
10 No    Yes 0         0         
11 No    Yes 1         0         
12 Yes   Yes 0         1         
13 Yes   No  0         0         
14 Yes   Yes 0         1         
15 Yes   Yes 0         1         
16 No    No  0         0         
17 Yes   No  0         1         
18 Yes   Yes 0         1         
19 No    Yes 0         1         
20 Yes   Yes 0         0

Also, create two interaction variables:

- Price_Bad_Shelf = Price* Bad_Shelf
- Price_Good_Shelf = Price* Good_Shelf

In [3]:
C1<-C1 %>%
    mutate(Price_Bad_Shelf=Price*Bad_Shelf) %>%
    mutate(Price_Good_Shelf=Price*Good_Shelf)
head(C1,20)

Sales CompPrice Income Advertising Population Price ShelveLoc Age Education
1   9.50 138        73    11          276        120   Bad       42  17       
2  11.22 111        48    16          260         83   Good      65  10       
3  10.06 113        35    10          269         80   Medium    59  12       
4   7.40 117       100     4          466         97   Medium    55  14       
5   4.15 141        64     3          340        128   Bad       38  13       
6  10.81 124       113    13          501         72   Bad       78  16       
7   6.63 115       105     0           45        108   Medium    71  15       
8  11.85 136        81    15          425        120   Good      67  10       
9   6.54 132       110     0          108        124   Medium    76  10       
10  4.69 132       113     0          131        124   Medium    76  17       
11  9.01 121        78     9          150        100   Bad       26  10       
12 11.96 117        94     4          503         94   Good      50  13       
13  3.98 122        35     2          393        136   Medium    62  18       
14 10.96 115        28    11           29         86   Good      53  18       
15 11.17 107       117    11          148        118   Good      52  18       
16  8.71 149        95     5          400        144   Medium    76  18       
17  7.58 118        32     0          284        110   Good      63  13       
18 12.29 147        74    13          251        131   Good      52  10       
19 13.91 110       110     0          408         68   Good      46  17       
20  8.73 129        76    16           58        121   Medium    69  12       
   Urban US  Bad_Shelf Good_Shelf Price_Bad_Shelf Price_Good_Shelf
1  Yes   Yes 1         0          120               0             
2  Yes   Yes 0         1            0              83             
3  Yes   Yes 0         0            0               0             
4  Yes   Yes 0         0            0               0             
5  Yes   No  1         0          128               0             
6  No    Yes 1         0           72               0             
7  Yes   No  0         0            0               0             
8  Yes   Yes 0         1            0             120             
9  No    No  0         0            0               0             
10 No    Yes 0         0            0               0             
11 No    Yes 1         0          100               0             
12 Yes   Yes 0         1            0              94             
13 Yes   No  0         0            0               0             
14 Yes   Yes 0         1            0              86             
15 Yes   Yes 0         1            0             118             
16 No    No  0         0            0               0             
17 Yes   No  0         1            0             110             
18 Yes   Yes 0         1            0             131             
19 No    Yes 0         1            0              68             
20 Yes   Yes 0         0            0               0

** Please estimate a linear regression model (using the lm function) with Sales as the dependent variable and Price as the independent variable.<br> **

In [4]:
lm.1<-lm(formula=Sales~Price,data=C1)
summary(lm.1)


Call:
lm(formula = Sales ~ Price, data = C1)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.5224 -1.8442 -0.1459  1.6503  7.5108 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 13.641915   0.632812  21.558   <2e-16 ***
Price       -0.053073   0.005354  -9.912   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.532 on 398 degrees of freedom
Multiple R-squared:  0.198,	Adjusted R-squared:  0.196 
F-statistic: 98.25 on 1 and 398 DF,  p-value: < 2.2e-16


- **  What is this model’s R-squared value?  0.198 **
- ** What is this model’s adjusted R-squared value? 0.196 **
- ** What is the estimated coefficient of Price? -0.053073 **
- ** What is the t- value of the coefficient of Price? -9.912 **
- ** Is the estimated coefficient for price statistically speaking different from 0 (at a significance level of 5%) <br>Yes, the p-value is less than 0.05 **

** Please estimate a linear regression model (using the lm function) with Sales as the dependent variable and Price, Bad_Shelf, and Good_Shelf as independent variables. ** 

In [5]:
lm.2<-lm(formula=Sales~Price+Bad_Shelf+Good_Shelf,data=C1)
summary(lm.2)


Call:
lm(formula = Sales ~ Price + Bad_Shelf + Good_Shelf, data = C1)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.8229 -1.3930 -0.0179  1.3868  5.0780 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 13.863824   0.487021  28.467  < 2e-16 ***
Price       -0.056698   0.004059 -13.967  < 2e-16 ***
Bad_Shelf   -1.862022   0.234748  -7.932 2.23e-14 ***
Good_Shelf   3.033825   0.245178  12.374  < 2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.917 on 396 degrees of freedom
Multiple R-squared:  0.5426,	Adjusted R-squared:  0.5391 
F-statistic: 156.6 on 3 and 396 DF,  p-value: < 2.2e-16


- ** What is this model’s coefficient for the variable "Price"? -0.056698 **

- ** What is this model’s coefficient for the variable “Bad_Shelf”?   -1.862022  **

- ** What is this model’s coefficient for the variable “Good_Shelf”? 3.033825 **

- ** What is the average value of Sales when Price = 0 for ShelveLoc = “Medium”?<br> Base case:Value of intercept=13.863824 **

- ** What is the average value of Sales when Price = 0 for ShelveLoc = “Bad” **


In [6]:
13.863824-1.862022

[1] 12.0018

- ** What is the average value of Sales when Price = 0 for ShelveLoc = “Good”? **

In [7]:
13.863824+3.033825

[1] 16.89765

** Please estimate a linear regression model (using the lm function) with Sales as the dependent variable and Price and ShelveLoc as independent variables.  **

In [8]:
lm.3<- lm(formula=Sales~Price+ShelveLoc,data=C1)
summary(lm.3)


Call:
lm(formula = Sales ~ Price + ShelveLoc, data = C1)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.8229 -1.3930 -0.0179  1.3868  5.0780 

Coefficients:
                 Estimate Std. Error t value Pr(>|t|)    
(Intercept)     12.001802   0.503447  23.839  < 2e-16 ***
Price           -0.056698   0.004059 -13.967  < 2e-16 ***
ShelveLocGood    4.895848   0.285921  17.123  < 2e-16 ***
ShelveLocMedium  1.862022   0.234748   7.932 2.23e-14 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.917 on 396 degrees of freedom
Multiple R-squared:  0.5426,	Adjusted R-squared:  0.5391 
F-statistic: 156.6 on 3 and 396 DF,  p-value: < 2.2e-16


- ** What is this model’s R-square value?  0.5426 **
- ** What is the average value of Sales when Price = 0 for ShelveLoc = “Bad”? Base case=12.001802   **
- ** What is the average value of Sales when Price = 0 for ShelveLoc = “Medium”?  **

In [9]:
12.001802+1.862022   

[1] 13.86382

- ** What is the average value of Sales when Price = 0 for ShelveLoc = “Good” **

In [10]:
12.001802 + 4.895848   

[1] 16.89765

- ** What is the estimated Coefficient of Price?  -0.056698 **

** Please estimate a linear regression model (using the lm function) with Sales as the dependent variable and Price, Bad_Shelf, Good_Shelf , Price_Bad_Shelf, and Price_Good_Shelf as independent variables **

In [11]:
lm.4<-lm(formula=Sales~Price+Bad_Shelf+Good_Shelf+Price_Bad_Shelf+Price_Good_Shelf,data=C1)
summary(lm.4)


Call:
lm(formula = Sales ~ Price + Bad_Shelf + Good_Shelf + Price_Bad_Shelf + 
    Price_Good_Shelf, data = C1)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.9037 -1.3461 -0.0595  1.3679  4.9037 

Coefficients:
                  Estimate Std. Error t value Pr(>|t|)    
(Intercept)      13.463465   0.663278  20.298  < 2e-16 ***
Price            -0.053236   0.005624  -9.465  < 2e-16 ***
Bad_Shelf        -1.630481   1.171616  -1.392 0.164813    
Good_Shelf        4.505399   1.202999   3.745 0.000207 ***
Price_Bad_Shelf  -0.001984   0.010007  -0.198 0.842907    
Price_Good_Shelf -0.012549   0.010050  -1.249 0.212541    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.918 on 394 degrees of freedom
Multiple R-squared:  0.5444,	Adjusted R-squared:  0.5386 
F-statistic: 94.17 on 5 and 394 DF,  p-value: < 2.2e-16


- ** What is this model’s base case? Medium Shelf  ** 
- ** When price = 0, is the average sales for products in bad shelf significantly different compared to the base case<br> No as Bad_Shelf is not statistically significant **
- ** For a unit change in price, does the average sales differ significantly between products in good shelf compared to the base case?<br> No as  Price_Good_Shelf  is not statistically significant **
